### PR TITLE
fix: client side routing when routing to same route, different params

### PR DESCRIPTION
**Pull request by eddieferrer**
VUE-1245

I did not like the solution of using `beforeRouteUpdate` because then the onus would be on the developer to add that hook and prefetch the components all the way down if a route with a param could possibly be navigated client side with a change in params. 

This solution leaves navigation prefetch behavior as it was unless, the route is the same and the params are different

### CODE REVIEW

File: src/client-entry.js [hunk 1]
```diff
@@ -135,12 +135,22 @@ router.onReady(() => {
 		// get newly activated components
 		const matched = router.getMatchedComponents(to);
 		const prevMatched = router.getMatchedComponents(from);
-		const activated = _dropWhile(matched, (c, i) => prevMatched[i] === c);
+		const areRoutesTheSame = JSON.stringify(to?.matched?.[0]?.path) === JSON.stringify(from?.matched?.[0]?.path);
+		const areParamsTheSame = JSON.stringify(to?.params) === JSON.stringify(from?.params);
```
Comment (dyersituations) [inline src/client-entry.js:139]: Do we need to check if params are not in the same order?

Comment (eddieferrer) [reply src/client-entry.js:139]: vue router returns them in the order they are defined in the route. So for a fictional route of 
`/lend-by-category/:category/:test` they always come back as {category: "foo", test: "bar"}

Comment (dyersituations) [reply src/client-entry.js:139]: Sounds good.

**Review by dyersituations — APPROVED** [review]

**Review by mcstover — APPROVED** [review]
While I don't believe this fully addresses re-fetching global components it _does_ focus to certain parts of the site. Probably lend-by-cat + settings sections at this time.

**Review by emuvente — APPROVED** [review]

**Pull request merged**